### PR TITLE
fix: Malformed JSON in `whereMatchesQuery`

### PR DESCRIPTION
**Pull request by juriSacchetta**
## Pull Request

- Report security issues [confidentially](https://github.com/parse-community/Parse-SDK-Flutter/security/policy).
- Any contribution is under this [license](https://github.com/parse-community/Parse-SDK-Flutter/blob/master/LICENSE).
- Link this pull request to an [issue](https://github.com/parse-community/Parse-SDK-Flutter/issues?q=is%3Aissue).

## Issue

Re-open PR: #943
Closes: #939

## Approach

Fix bug in JSON creation when `whereMatchesQuery` is used.

## Tasks

- [x] Add tests
- [ ] Add changes to documentation (guides, repository pages, code comments)

### CODE REVIEW

**Review by mbfakourii — APPROVED** [review]
Great, thanks for the test

**Review by mtrezza — CHANGES_REQUESTED** [review]
Could you add a changelog entry?

**Comment by juriSacchetta** [comment]
> Could you add a changelog entry?

I don't know how to do it, can you give me some indication?

**Comment by mbfakourii** [comment]
> > Could you add a changelog entry?
> 
> I don't know how to do it, can you give me some indication?

An example is in this [PR.](https://github.com/parse-community/Parse-SDK-Flutter/pull/811/commits/d48de7035ea9746f39a4f42652504da7ff357e83).

It means is adding changes and bump version (increasing the package version).

**Comment by juriSacchetta** [comment]
> > > Could you add a changelog entry?
> > 
> > 
> > I don't know how to do it, can you give me some indication?
> 
> An example is in this [PR](https://github.com/parse-community/Parse-SDK-Flutter/pull/811/commits/d48de7035ea9746f39a4f42652504da7ff357e83).
> 
> It means is adding changes and bump version (increasing the package version).

Thank you for your reply. Please let me know if it's alright now.

**Comment by mbfakourii** [comment]
> > > > Could you add a changelog entry?
> > > 
> > > 
> > > I don't know how to do it, can you give me some indication?
> > 
> > 
> > An example is in this [PR](https://github.com/parse-community/Parse-SDK-Flutter/pull/811/commits/d48de7035ea9746f39a4f42652504da7ff357e83).
> > It means is adding changes and bump version (increasing the package version).
> 
> Thank you for your reply. Please let me know if it's alright now.

Looks good

File: packages/dart/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,9 @@
+## [5.1.3](https://github.com/parse-community/Parse-SDK-Flutter/compare/dart-5.1.2...dart-5.1.3) (2023-05-29)
```
Comment (mtrezza) [inline packages/dart/CHANGELOG.md:1]: Corrected the changelog date, assuming we release today.

@mbfakourii Could you make a formal review?

```suggestion
## [5.1.3](https://github.com/parse-community/Parse-SDK-Flutter/compare/dart-5.1.2...dart-5.1.3) (2023-07-18)
```

**Review by mtrezza — CHANGES_REQUESTED** [review]

File: packages/dart/CHANGELOG.md (hunk 1 above)
Comment (mbfakourii) [reply packages/dart/CHANGELOG.md:1]: I checked, there is no problem

Comment (mtrezza) [reply packages/dart/CHANGELOG.md:1]: I mean, could you make a review on GitHub?

**Review by mbfakourii — APPROVED** [review]

File: packages/dart/CHANGELOG.md (hunk 1 above)
Comment (mbfakourii) [reply packages/dart/CHANGELOG.md]: sure

**Review by mtrezza — APPROVED** [review]

**Pull request merged**